src/cart/src/cart/crt_context.c

```diff
@@ -1293,37 +1267,83 @@ crt_progress(crt_context_t crt_ctx, int64_t timeout,
 			hg_timeout = timeout;
 	}
 
-	while (true) {
+	/** loop until callback returns non-null value */
+	while ((rc = cond_cb(arg)) == 0) {
```

Comment:
on line 1234 we assume cond_cb is optional argument, but here we assume it is always set. either we need check at top to error if cond_cb is not set and remove conditional check on line 1234, or have logic here that handles cond_cb being null

Reply:
Forgot to remove the conditional check.